pubsub/topic.go

```diff
@@ -592,6 +592,189 @@ func (i *IngestionDataSourceCloudStoragePubSubAvroFormat) isCloudStorageIngestio
+type EventHubsState int
+
+const (
+	// EventHubsStateUnspecified is the default value. This value is unused.
+	EventHubsStateUnspecified = iota
+
+	// EventHubsStateActive means the state is active.
+	EventHubsStateActive
+
+	// EventHubsStatePermissionDenied indicates encountered permission denied error
+	// while consuming data from Event Hubs.
+	// This can happen when `client_id`, or `tenant_id` are invalid. Or the
+	// right permissions haven't been granted.
+	EventHubsStatePermissionDenied
+
+	// EventHubsStatePublishPermissionDenied indicates permission denied encountered
+	// while publishing to the topic.
+	EventHubsStatePublishPermissionDenied
+
+	// EventHubsStateNamespaceNotFound indicates the provided Event Hubs namespace couldn't be found.
+	EventHubsStateNamespaceNotFound
+
+	// EventHubsStateNotFound indicates the provided Event Hub couldn't be found.
+	EventHubsStateNotFound
+
+	// EventHubsStateSubscriptionNotFound indicates the provided Event Hubs subscription couldn't be found.
+	EventHubsStateSubscriptionNotFound
+
+	// EventHubsStateResourceGroupNotFound indicates the provided Event Hubs resource group couldn't be found.
+	EventHubsStateResourceGroupNotFound
+)
+
+// IngestionDataSourceAzureEventHubs are ingestion settings for Azure Event Hubs.
+type IngestionDataSourceAzureEventHubs struct {
+	// Output only field that indicates the state of the Event Hubs ingestion source.
+	State EventHubsState
+
+	// Name of the resource group within the Azure subscription
+	ResourceGroup string
+
+	// Name of the Event Hubs namespace
+	Namespace string
+
+	// Rame of the Event Hub.
+	EventHub string
+
+	// Client ID of the Azure application that is being used to authenticate Pub/Sub.
+	ClientID string
+
+	// Tenant ID of the Azure application that is being used to authenticate Pub/Sub.
+	TenantID string
+
+	// The Azure subscription ID
+	SubscriptionID string
+
+	// GCPServiceAccount is the GCP service account to be used for Federated Identity
+	// authentication.
+	GCPServiceAccount string
+}
+
+var _ IngestionDataSource = (*IngestionDataSourceAzureEventHubs)(nil)
+
+func (i *IngestionDataSourceAzureEventHubs) isIngestionDataSource() bool {
+	return true
+}
+
+// MSKState denotes the possible states for ingestion from Amazon MSK.
+type MSKState int
+
+const (
+	// MSKStateUnspecified is the default value. This value is unused.
+	MSKStateUnspecified = iota
+
+	// MSKActive indicates MSK topic is active.
+	MSKActive
+
+	// MSKPermissionDenied indicates permission denied encountered while consuming data from Amazon MSK.
+	MSKPermissionDenied
+
+	// MSKPublishPermissionDenied indicates permission denied encountered while publishing to the topic.
+	MSKPublishPermissionDenied
+
+	// MSKClusterNotFound indicates the provided Msk cluster wasn't found.
+	MSKClusterNotFound
+
+	// MSKTopicNotFound indicates the provided topic wasn't found.
+	MSKTopicNotFound
+)
+
+// IngestionDataSourceAmazonMSK are ingestion settings for Amazon MSK.
```

Comment:
AWSMSK would more closely match our proto, but does look a bit strange. I would be fine keeping AmazonMSK or maybe AwsMsk.

Reply:
Go style guide says initialisms must keep the same case, so unfortunately `AwsMsk` isn't correct. I think AmazonMSK is clearer, and matches the official product name ("Amazon Managed Streaming for Apache Kafka") anyway

Reply:
AmazonMSK is good with me.